### PR TITLE
[732] If all devices ordered page

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/tEnKg7FQ/732-build-all-devices-ordered-page-for-schools

### Changes proposed in this pull request

- This is displayed instead of the cannot order page when the user can order but have reached their cap

### Screenshots

image

### Guidance to review

- Login as a school user
- The school can order devices but cap == allocation
- Attempt to user
- Should see the new page instead of the cannot order page

### CODE REVIEW

File: app/controllers/school/devices_controller.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ def order
       end
     elsif @school.can_order_for_specific_circumstances? && @school.can_order_devices?
       render :can_order_for_specific_circumstances
+    elsif @school.can_order?
```
Comment (benilovj) [inline app/controllers/school/devices_controller.rb:13]: I'm not able to follow the logic in this `if ... else ...` block without looking at the template names.

`@school.can_order?` and `@school.can_order_devices?` are a particular source of confusion. Would you be able to rename them to something longer but hopefully more explicit?

Comment (asmega) [reply app/controllers/school/devices_controller.rb]: Agreed theres quite a few permutations and getting more confusing. I've made it more explicit even though it's not actually needed

File: spec/controllers/school/devices_controller_spec.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@
 
         it 'renders can_order' do
```
Comment (benilovj) [inline spec/controllers/school/devices_controller_spec.rb:44]: this needs to be updated as it no longer matches the expectation...

File: app/controllers/school/devices_controller.rb (hunk 1 above)
Comment (benilovj) [reply app/controllers/school/devices_controller.rb]: thanks. I still think that this logic needs an overhaul because:

1. there's a bunch of similar-sounding methods that do similar things and are quite hard to follow
2. there's too much logic in controllers that needs to be pushed down to the model or service level

however that probably shouldn't block this change. I will pick it up separately as I have some ideas.

**Review by benilovj — APPROVED** [review]
Fine for now, except the spec description that needs to be updated.

**Review by benilovj — CHANGES_REQUESTED** [review]
I'm not sure this will display the right thing when the school has placed all their orders for specific circumstances...

File: spec/controllers/school/devices_controller_spec.rb (hunk 2 above)
Comment (asmega) [reply spec/controllers/school/devices_controller_spec.rb]: fixed up now

**Review by benilovj — APPROVED** [review]
